### PR TITLE
Drop support for legacy Python 2.7

### LINKED ISSUE #234: Drop Python 2.7 support going forward?
Is it worth it to drop Py2.7 support going forward and make the current release LTS? I believe so, as at least NumPy has done the same, and we might end up needing 1.17 features.

cc @rgommers and anyone else interested.

### PR DESCRIPTION
**Pull request by hugovk**
Fixes #234.

And upgrade Python syntax using https://github.com/asottile/pyupgrade.

### CODE REVIEW

**Comment by hameerabbasi** [comment]
This pull request **fixes 4 alerts** when merging 6969cd59212b4569a1cd855c9b57c471a0f7ddf0 into 4b262fb7ea6bdb13324031179e76a3c0a5b3fc5a - [view on LGTM.com](https://lgtm.com/projects/g/pydata/sparse/rev/pr-e1159d863cffed6ce65400ebfe5723ba6420887d)

**fixed alerts:**

* 4 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: setup.py [hunk 1]
```diff
@@ -56,11 +56,10 @@
         'Operating System :: OS Independent',
         'License :: OSI Approved :: BSD License',
         'Programming Language :: Python',
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 2.7',
         'Programming Language :: Python :: 3',
         'Programming Language :: Python :: 3.5',
         'Programming Language :: Python :: 3.6',
+        'Programming Language :: Python :: Only',
```
Comment (hameerabbasi) [inline setup.py]: Shouldn’t this be `Python :: 3 :: Only`?

Comment (hugovk) [reply setup.py]: It should. Fixed it now, thanks!

https://pypi.org/pypi?%3Aaction=list_classifiers

**Comment by hameerabbasi** [comment]
Could you also add testing for 3.7?

File: .travis.yml [hunk 2]
```diff
@@ -8,6 +8,8 @@ matrix:
     - python: 3.5
       env: NUMPY_VERSION="<1.14.0"
     - python: 3.6
+    - python: 3.7
+      dist: xenial
```
Comment (hameerabbasi) [inline .travis.yml]: I wasn't aware that you needed this, thanks!

Comment (hugovk) [reply .travis.yml]: Yes, see https://github.com/travis-ci/travis-ci/issues/9069 / https://github.com/travis-ci/travis-ci/issues/9815 for details (`sudo` is no longer needed).

**Comment by hameerabbasi** [comment]
I don't mean to ask for too much, but it seems like there are `%`-strings in the code for a while now. Is there a way to automatically replace those? (According to the `pyupgrade` docs, `--py3-plus` should have done it).

**Comment by hugovk** [comment]
Maybe pyupgrade only replaces `%` formatters on explicit strings.

**Comment by hameerabbasi** [comment]
Understood, I'll merge once the last check passes.

**Comment by hameerabbasi** [comment]
This pull request **fixes 4 alerts** when merging ce5b2c637a3c2d36e96e3ca8165854602c6504a8 into 4b262fb7ea6bdb13324031179e76a3c0a5b3fc5a - [view on LGTM.com](https://lgtm.com/projects/g/pydata/sparse/rev/pr-866759231b57578fbff203da7dcb215bc74e1596)

**fixed alerts:**

* 4 for Unused import

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by hameerabbasi** [comment]
Merging, thanks a lot, @hugovk!

**Pull request merged**